codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -864,6 +911,19 @@ public DefaultStream reservePushStream(int streamId, Http2Stream parent) throws
         private void addStream(DefaultStream stream) {
             // Add the stream to the map and priority tree.
             streamMap.put(stream.id(), stream);
+
+            // Initialize the flow state
+            final Http2RemoteFlowController remoteFlowController = remote().flowController();
```

Comment:
I had this and the `local().flowController()` call inlined before...but the exception was much more nebulous as to what is going on.  I can change back if we think this null checking is too verbose.